src/api_server/src/lib.rs

```diff
@@ -30,30 +29,18 @@ use vmm::rpc_interface::{VmmAction, VmmActionError, VmmData};
 use vmm::vmm_config::snapshot::SnapshotType;
 
 use crate::parsed_request::{ParsedRequest, RequestAction};
+use crate::Error::ServerCreation;
 
 /// Shorthand type for a request containing a boxed VmmAction.
 pub type ApiRequest = Box<VmmAction>;
 /// Shorthand type for a response containing a boxed Result.
 pub type ApiResponse = Box<std::result::Result<VmmData, VmmActionError>>;
 
 /// Errors thrown when binding the API server to the socket path.
-#[derive(thiserror::Error)]
+#[derive(Debug)]
 pub enum Error {
-    /// IO related error.
-    #[error("IO error: {0}")]
-    Io(io::Error),
-    /// EventFD related error.
-    #[error("EventFd error: {0}")]
-    Eventfd(io::Error),
-}
-
-impl fmt::Debug for Error {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        match *self {
-            Error::Io(ref err) => write!(f, "IO error: {}", err),
-            Error::Eventfd(ref err) => write!(f, "EventFd error: {}", err),
-        }
-    }
+    /// HTTP Server creation related error.
+    ServerCreation(ServerError),
```

Comment:
Nit: Errors types with a single variant should ideally be newtypes.
```rust
/// HTTP Server creation related error.
#[derive(Debug)]
pub struct ServerCreationError(ServerError)
```

Reply:
I do plan on adding more errors in the future (you can notice that there is a `panic` below  when trying to apply the seccomp filters -> I think we need to get rid of and  propagate and handle it the same we do for errors triggered by `ServerCreation`).